portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -83,18 +131,30 @@ private String convertToJsonUsingGson(List comments) {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-    // Get the input from the form and a currenttimestamp
+
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Only logged-in users can post messages
+    if (!userService.isUserLoggedIn()) {
+      String loginUrl = userService.createLoginURL(request.getHeader("referer"));
+      response.sendRedirect(loginUrl);
+      return;
+    }
+
+    // Get the input from the form and a current timestamp
     String text = request.getParameter("text-input");
     long timestamp = System.currentTimeMillis();
+    String email = userService.getCurrentUser().getEmail();
```

Comment:
emailAddress is a better name, I wonder what are the rules regarding storing it... do you need user's consent?

Reply:
Changed the name :)
That's a great question. A previous iteration had us store and display the entire email address, with an optional extension to have a nickname instead. Personally I don't like the idea of storing the entire email, so here I just access it and store the first four characters as a pseudo-nickname